File: java/poker/src/main/java/com/duchen/poker/model/type/SingleTypeChecker.java
package com.duchen.poker.model.type;

import java.util.Arrays;
import java.util.List;

public class SingleTypeChecker implements TypeChecker {

    @Override
    public CardGroupType getType() {
        return CardGroupType.SINGLE;
    }

    @Override
    public List<Integer> getMyPossibleCardCount() {
        return Arrays.asList(1);
    }

    @Override
    public int getMaxCardIfIsMyType(List<Integer> cards) {
        return cards.get(0);
    }
}

File: java/poker/src/main/kotlin/poker/model/type/SingleTypeChecker.kt
package poker.model.type

import java.util.Arrays

class SingleTypeChecker : TypeChecker {

    override val type: CardGroupType
        get() = CardGroupType.SINGLE

    override val myPossibleCardCount: List<Int>
        get() = Arrays.asList(1)

    override fun getMaxCardIfIsMyType(cards: MutableList<Int>): Int {
        return cards[0]
    }
}
